source/common/config/new_grpc_mux_impl.cc

```diff
@@ -130,24 +117,42 @@ Watch* NewGrpcMuxImpl::addWatch(const std::string& type_url, const std::set<std:
 void NewGrpcMuxImpl::updateWatch(const std::string& type_url, Watch* watch,
                                  const std::set<std::string>& resources) {
   ASSERT(watch != nullptr);
-  auto sub = subscriptions_.find(type_url);
-  RELEASE_ASSERT(sub != subscriptions_.end(),
-                 fmt::format("Watch of {} has no subscription to update.", type_url));
-  auto added_removed = sub->second->watch_map_.updateWatchInterest(watch, resources);
-  sub->second->sub_state_.updateSubscriptionInterest(added_removed.added_, added_removed.removed_);
+  SubscriptionState* sub = subscriptionStateFor(type_url);
+  WatchMap& watch_map = watchMapFor(type_url);
+
+  auto added_removed = watch_map.updateWatchInterest(watch, resources);
+  sub->updateSubscriptionInterest(added_removed.added_, added_removed.removed_);
+
   // Tell the server about our change in interest, if any.
-  if (sub->second->sub_state_.subscriptionUpdatePending()) {
+  if (sub->subscriptionUpdatePending()) {
     trySendDiscoveryRequests();
   }
 }
 
 void NewGrpcMuxImpl::addSubscription(const std::string& type_url,
                                      std::chrono::milliseconds init_fetch_timeout) {
-  subscriptions_.emplace(type_url, std::make_unique<SubscriptionStuff>(type_url, init_fetch_timeout,
-                                                                       dispatcher_, local_info_));
+  watch_maps_.emplace(type_url, std::make_unique<WatchMap>());
+  subscriptions_.emplace(type_url, subscription_state_factory_->makeSubscriptionState(
+                                       type_url, *watch_maps_[type_url], init_fetch_timeout));
   subscription_ordering_.emplace_back(type_url);
 }
 
+SubscriptionState* NewGrpcMuxImpl::subscriptionStateFor(const std::string& type_url) {
```

Comment:
Since it's not nullptr: should the return type SubscriptionState&?

Reply:
Done.